apis/v1alpha2/policy_types.go

```diff
@@ -31,12 +31,29 @@ const (
 	PolicyLabelKey = "gateway.networking.k8s.io/policy"
 )
 
-// PolicyTargetReference identifies an API object to apply a direct or
+// LocalPolicyTargetReference identifies an API object to apply a direct or
 // inherited policy to. This should be used as part of Policy resources
 // that can target Gateway API resources. For more information on how this
 // policy attachment model works, and a sample Policy resource, refer to
 // the policy attachment documentation for Gateway API.
-type PolicyTargetReference struct {
+type LocalPolicyTargetReference struct {
+	// Group is the group of the target resource.
+	Group Group `json:"group"`
+
+	// Kind is kind of the target resource.
+	Kind Kind `json:"kind"`
+
+	// Name is the name of the target resource.
+	Name ObjectName `json:"name"`
+}
+
+// NamespacedPolicyTargetReference identifies an API object to apply a direct or
+// inherited policy to, potentially in a different namespace. This should only
+// be used as part of Policy resources that need to be able to target resources
+// in different namespaces. For more information on how this policy attachment
+// model works, and a sample Policy resource, refer to the policy attachment
+// documentation for Gateway API.
+type NamespacedPolicyTargetReference struct {
```

Comment:
AFAIK this is not referenced elsewhere  , only in this `gwctl` tool, is that correct?

Reply:
Yep, it's part of the types that we encourage policy extensions to use, but not used internally.